test/apps/iodemo/io_demo.cc

```diff
@@ -650,19 +650,25 @@ class P2pDemoCommon : public UcxContext {
             }
         }
 
-        void init(size_t data_size, void *ext_buf)
+        void init(size_t data_size, void *ext_buf, bool validate)
         {
             assert(ext_buf != NULL);
             assert(_iov.empty());
             assert(_extra_buf == NULL);
 
             _data_size = data_size;
             _extra_buf = ext_buf;
+            _validate  = validate;
         }
 
         inline Buffer &operator[](size_t i) const
         {
-            return *_iov[i];
+            if (!_iov.empty()) {
+                return *_iov[i];
+            } else {
+                assert(_extra_buf != NULL);
```

Comment:
why need this assert?
just return `(!_iov.empty()) ? *_iov[i] : _bogus_buf`

Reply:
done